Modules/Shop/Http/Services/ShopService.php

```diff
@@ -45,31 +45,28 @@ public function getBrandProducts($request): array
         if ($brandDetails) {
             $userId = $brandDetails->user_id;
             //all products count
-            $allProductsCount = Product::where('user_id', $userId)->where('status', 'publish')->count();
-
+            $allProductsCount = User::find($userId)->getAllProducts()->where('status', 'publish')->count();
             //new products count
-            $newProductsCount = Product::where('user_id', $userId)->where('status', 'publish')->where('created_at', '>', now()->subDays(7)->endOfDay())->count();
-
-            $productCategories = Product::select(DB::raw("count(*) as prdct_count"), "category")->where('status', 'publish')->where('user_id', $userId)->groupBy('category')->get();
+            $newProductsCount = User::find($userId)->getAllProducts()->where('status', 'publish')->where('created_at', '>', now()->subDays(7)->endOfDay())->count();
 
+            $productCategories = User::find($userId)->getAllProducts()->selectRaw('count(*) as prdct_count, category, categories.title, categories.slug, categories.parent_id')->leftJoin('categories', 'products.category', '=', 'categories.id')->where('products.status', 'publish')->groupBy('products.category')->get();
```

Comment:
Update these three queries with this query
$user = User::with(['getAllProducts' => function ($query) {
    $query->select('category', 'created_at')
        ->where('status', 'publish')
        ->withCount(['category'])
        ->leftJoin('categories', 'products.category', '=', 'categories.id')
        ->where('categories.status', 'publish')
        ->groupBy('products.category', 'categories.title', 'categories.slug', 'categories.parent_id');
}])->find($userId);

$allProductsCount = $user->getAllProducts->count();

$newProductsCount = $user->getAllProducts
    ->where('created_at', '>', now()->subDays(7)->endOfDay())
    ->count();

$productCategories = $user->getAllProducts;
**Make sure to test it before pushing and try to follow this approach next.**